src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +156,64 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    /// <summary>
+    /// Checks the merge element for a frequency attribute. Then determines whether the current run
+    /// matches the frequency criteria. Valid frequency values are 'daily' and 'weekly'.
+    /// </summary>
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // Throw when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            throw new Exception($"Unexpected merge frequency specified: '{frequency}'. Valid values are 'daily' and 'weekly'.");
+        }
+
+        // Since this is run on AzDO as an automated cron pipeline, times are in UTC.
+        // See https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#scheduled-triggers
+        var runDateTime = DateTime.UtcNow;
```

Comment:
Please get DateTime.UtcNow only once in a run of the tool. e.g. by passing in a `DateTime currentTime` parameter.

Reply:
Doh! good catch